source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -3,17 +3,53 @@
 #include "source/common/common/assert.h"
 #include "source/common/common/enum_to_int.h"
 #include "source/common/common/logger.h"
+#include "source/common/config/xds_resource.h"
 #include "source/common/http/codes.h"
+#include "source/common/http/utility.h"
+#include "source/extensions/filters/http/well_known_names.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace OnDemand {
 
+namespace {
+
+template <typename ProtoConfig>
+Upstream::OdCdsApiHandlePtr createOdCdsApi(const ProtoConfig& proto_config,
+                                           Upstream::ClusterManager& cm,
+                                           ProtobufMessage::ValidationVisitor& validation_visitor) {
+  if (!proto_config.has_odcds_config() && proto_config.resources_locator().empty()) {
+    return nullptr;
+  }
+  if (proto_config.resources_locator().empty()) {
+    // TODO: Would be nice if OptRef accepted absl::null_opt.
+    OptRef<xds::core::v3::ResourceLocator> null_opt_ref;
```

Comment:
As I understand this, OptRef is an optional reference, so either it should refer to a valid object, or be `absl::nullopt`

Reply:
IIRC, the comment was about OptRef having no constructor that accepts `abs::nullopt_t`, so I couldn't pass `absl::nullopt` to a function taking `OptRef<T>`.

Reply:
I'd drop this comment, or move it to the `OptRef` interface comments (or just fix, it should be a very small change).

Reply:
I can fix that. Didn't do it before because I was scared that it would rebuild half of the world. :)